src/Common/src/System/Security/Cryptography/Asn1V2.cs

```diff
@@ -0,0 +1,5246 @@
+                
+                if (tag.TagClass != TagClass.Universal)
+                {
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                }
+
+                totalRead += headerLength;
+
+                if (isIndefinite && tag.TagValue == (int)UniversalTagNumber.EndOfContents)
+                {
+                    ValidateEndOfContents(tag, length, headerLength);
+
+                    bytesRead = totalRead;
+                    return totalContent;
+                }
+
+                if (tag.TagValue != (int)UniversalTagNumber.OctetString)
+                {
+                    // T-REC-X.690-201508 sec 8.7.3.2 (in particular, Note 2)
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                }
+
+                if (ruleSet == AsnEncodingRules.CER)
+                {
+                    if (tag.IsConstructed || lastSegmentLength != MaxCERSegmentSize)
+                    {
+                        // T-REC-X.690-201508 sec 9.2
+                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                    }
+                }
+
+                cur = cur.Slice(headerLength);
+
+                if (length == null)
+                {
+                    totalContent += CopyConstructedOctetString(
+                        cur,
+                        ref destination,
+                        write,
+                        ruleSet,
+                        true,
+                        ref lastSegmentLength,
+                        out int nestedBytesRead);
+
+                    totalRead += nestedBytesRead;
+                    cur = cur.Slice(nestedBytesRead);
+                }
+                else if (tag.IsConstructed)
+                {
+                    totalContent += CopyConstructedOctetString(
+                        Slice(cur, 0, length.Value),
+                        ref destination,
+                        write,
+                        ruleSet,
+                        false,
+                        ref lastSegmentLength,
+                        out int nestedContentRead);
+
+                    totalRead += nestedContentRead;
+                    cur = cur.Slice(nestedContentRead);
+                }
+                else
+                {
+                    int lengthValue = length.Value;
+
+                    // T-REC-X.690-201508 sec 9.2
+                    if (ruleSet == AsnEncodingRules.CER && lengthValue > MaxCERSegmentSize)
+                    {
+                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                    }
+
+                    totalRead += lengthValue;
+                    totalContent += lengthValue;
+                    lastSegmentLength = lengthValue;
+
+                    ReadOnlyMemory<byte> segment = Slice(cur, 0, lengthValue);
+             
```

Comment:
Doesn't `BigInteger` have a "write this directly to a span as big-endian" API?

Reply:
> Doesn't BigInteger have a "write this directly to a span as big-endian" API?

Nope.  Not yet.  https://github.com/dotnet/corefx/pull/25324 will add one, though.  Then there's just a question of does this class need to work in `netstandard`-compiled targets, or just `netcoreapp` ones, which limits the amount of new API that can be called.